### PR TITLE
airbyte-lib: Connector documentation

### PR DESCRIPTION
**Pull request by flash1293**
Add capabilities to effectively document airbyte-lib-enabled connectors to our doc system:
* Check the metadata whether the connector is published on pypi and adds an icon to the availability row in the header decoration
* Introduce the components AirbyteLibExample and SpecSchema which can be used in documentation files to render a code sample on how to use the connector and a collapsible element describing the full config schema

The new components are picked up by a new remark plugin that's injecting the spec of the connector as a prop to the component so it can be used to render out a generated sample config / the spec schema.

This adds the following dependencies:
* `json-schema-faker` to generate an example config object only based on the schema
* `@headlessui/react ` for the collapsible implementation
* `sanitize-html` to safely render HTML from the schema in the docs

See the following connector doc pages for examples:
* Google Drive: Complex config
* Stripe: Medium config
* Apify Dataset: Simple config

### CODE REVIEW

**Comment by flash1293** [comment]
TODO: Reimplement json schema rendering

**Comment by flash1293** [comment]
@timroes @nataliekwong could I get a look from you on this? From the basic styling questions I'm going to meet on Tuesday with @nora-airbyte as well.

**Comment by flash1293** [comment]
Thanks @nataliekwong !

> fields are listed in order of appearance (I think), and that means that we list optional fields before required ones

We can actually control this, added some code to show the required ones first

> We also mix the schema configuration with the connector configuration settings, so a user is context switching between the two "types" of configuration options

Not sure I follow, could you elaborate on that point?

**Comment by nataliekwong** [comment]
Oh nice, thanks for reorganizing! I think it may have resolved itself (in this example, at least) with that change: 
Google-Drive-Airbyte-Documentation

I think previously the streams were listed between two other settings, but now it's at the top (Should it be at the bottom? That would then break the "list required fields at the top" rule). I think of the connector's overall configuration settings differently from the streams, because I'd expect the connector configuration to determine the streams I have available. But perhaps that's an incorrect assumption in the case of airbyte-lib. Anyway, following that assumption, I'd then expect to set the connector-level settings before defining specific streams.

**Comment by flash1293** [comment]
@timroes I adopted @nora-airbyte s design and made it more table-y:
Screenshot 2024-01-18 at 15 46 30

IMHO it looks pretty cool. Could you give it a review?

File: docs/integrations/sources/google-drive.md [hunk 1]
```diff
@@ -243,6 +243,18 @@ One record will be emitted for each document. Keep in mind that large files can
 
 Before parsing each document, the connector exports Google Document files to Docx format internally. Google Sheets, Google Slides, and drawings are internally exported and parsed by the connector as PDFs.
 
+<HideInUI>
+
+## Usage with airbyte-lib
+
+For document file type streams, make sure the `tesseract` and `pdftotext` libraries are installed. On MacOS, you can install them with `brew install poppler tesseract`.
```
Comment (timroes) [inline docs/integrations/sources/google-drive.md:250]: Why are we putting `MacOS` specifically here as an example to install those, but no other system?

**Comment by timroes** [comment]
Generally I feel this looks pretty good. The only thing irking me a bit is how we display arrays. We somehow nest a `items[x]` key once you expand the actual array, so it looks like the following:



Specifically for arrays of primitive datatypes like string[] this looks imho more confusing then helpful:



Is there any chance for simple arrays to render it as `array<string>` as a type and don't have the collapsed `items[x]`. Also for complex ones I think we should not have the separate `items[x]`, since imho it makes it more confusing and looks like the path would be `streams.items[0]` to access something. Could we just render them as `array<object>` and have the child properties directly nested when expanding the array?

File: docs/integrations/sources/google-drive.md (hunk 1 above)
Comment (flash1293) [reply docs/integrations/sources/google-drive.md:250]: This is just an example, I will remove it from this PR before merging (only the pure implementation of the components will go on, rolling it out to connectors on a separate PR)

File: docusaurus/src/components/AirbyteLibExample.jsx [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+config = ${JSON.stringify(fakeConfig, null, 2)}
+
+result = ab.get_connector(
+    "${connector}",
+    config=config,
+).read_all()
+
+for record in result.cache.streams["my_stream:name"]:
+  print(record)`} </CodeBlock>
+    <p>You can find more information in the airbyte_lib quickstart guide.</p>
```
[the remark this answers is not in the document]
Comment (flash1293) [reply docusaurus/src/components/AirbyteLibExample.jsx:32]: It doesn't yet - I will work on this on a separate PR. As mentioned, this component will not be rendered yet, in this PR I just want to get the first version of the components ready.

**Comment by flash1293** [comment]
Thanks @timroes !

Your suggestions for arrays make a lot of sense to me - I will implement that.

**Comment by flash1293** [comment]
@timroes Done - it's now showing the properties of the array items directly. I did a similar thing for constant values - they are rendered as the "type" now:
Screenshot 2024-01-19 at 17 40 28

**Comment by flash1293** [comment]
If you are OK with the basic component I will remove the actual usages so we can get the pre-requisite in.

**Review by timroes — APPROVED** [review]

**Pull request merged**